pkg/image/prune/imagepruner_test.go

```diff
@@ -663,6 +701,67 @@ func TestImagePruning(t *testing.T) {
 			expectedDeletions:      []string{"id"},
 			expectedUpdatedStreams: []string{},
 		},
+		"image exceeding limits": {
+			keepOverSizeLimitImages: newBool(false),
+			images: imageList(
+				unmanagedImage("id", "otherregistry/foo/bar@id", false, "", ""),
+				sizedImage("id2", registryURL+"/foo/bar@id2", 100),
+				sizedImage("id3", registryURL+"/foo/bar@id3", 200),
+			),
+			streams: streamList(
+				stream(registryURL, "foo", "bar", tags(
+					tag("latest",
+						tagEvent("id", "otherregistry/foo/bar@id"),
+						tagEvent("id2", registryURL+"/foo/bar@id2"),
+						tagEvent("id3", registryURL+"/foo/bar@id3"),
+					),
+				)),
+			),
+			expectedDeletions:      []string{"id3"},
+			expectedUpdatedStreams: []string{"foo/bar|id3"},
+			limits:                 map[string][]*kapi.LimitRange{"foo": limitList(100, 200)},
+		},
+		"image within allowed limits": {
+			keepOverSizeLimitImages: newBool(false),
+			images: imageList(
+				unmanagedImage("id", "otherregistry/foo/bar@id", false, "", ""),
+				sizedImage("id2", registryURL+"/foo/bar@id2", 100),
+				sizedImage("id3", registryURL+"/foo/bar@id3", 200),
+			),
+			streams: streamList(
+				stream(registryURL, "foo", "bar", tags(
+					tag("latest",
+						tagEvent("id", "otherregistry/foo/bar@id"),
+						tagEvent("id2", registryURL+"/foo/bar@id2"),
+						tagEvent("id3", registryURL+"/foo/bar@id3"),
+					),
+				)),
+			),
+			expectedDeletions:      []string{},
+			expectedUpdatedStreams: []string{},
+			limits:                 map[string][]*kapi.LimitRange{"foo": limitList(300)},
+		},
+		"image stream - keep most recent n images, even with exceed-limits true": {
+			keepOverSizeLimitImages: newBool(false),
+			images: imageList(
+				unmanagedImage("id", "otherregistry/foo/bar@id", false, "", ""),
+				image("id2", registryURL+"/foo/bar@id2"),
+				image("id3", registryURL+"/foo/bar@id3"),
+				image("id4", registryURL+"/foo/bar@id4"),
+			),
+			streams: streamList(
+				stream(registryURL, "foo", "bar", tags(
+					tag("latest",
+						tagEvent("id", "otherregistry/foo/bar@id"),
+						tagEvent("id2", registryURL+"/foo/bar@id2"),
+						tagEvent("id3", registryURL+"/foo/bar@id3"),
+						tagEvent("id4", registryURL+"/foo/bar@id4"),
+					),
+				)),
+			),
+			expectedDeletions:      []string{"id4"},
+			expectedUpdatedStreams: []string{"foo/bar|id4"},
+		},
 	}
```

Comment:
A test with multiple image streams in multiple namespaces with different size limits would be awesome. But I won't let the perfect be the enemy of good.